File: Latihan2_1/Product.java
// class Product is parent of class Hardware
public class Product {
    private String idProduct;
    private String price;
    
    // constructor kosong
    public Product() {
    }
    
    // constructor
    public Product(String price, String idProduct) {
        this.idProduct = idProduct;
        this.price = price;       
    }

    // set product idProduct
    public void set_idProduct(String idProduct) {
        this.idProduct = idProduct;
    }
    // get product idProduct
    public String get_idProduct() {
        return idProduct;
    }

    // set product price
    public void set_price(String price) {
        this.price = price;
    }
    // get product price
    public String get_price() {
        return price;
    }


}

File: Latihan2_2/cpp/Product.cpp
#include <iostream>
#include <string>
using namespace std;

class Product{
    // atribut private
    private:
        string price;
        string idProduct;
    
    public:
        //constructor  kosong     
        Product(){
        }
        
        //constructor  kosong     
        Product(string price, string idProduct){
            this->price = price;
            this->idProduct = idProduct;
        }
        
        // set memory price
        void set_price(string price){
            this->price = price;
        }
        // get memory price
        string get_price(){
            return this->price;
        }

        // set memory idProduct
        void set_idProduct(string idProduct){
            this->idProduct = idProduct;
        }
        // get memory product
        string get_idProduct(){
            return this->idProduct;
        }

        // destructor
        ~Product(){
        }
};